docs/puzzles/day16.md

```diff
@@ -1,11 +1,572 @@
 import Solver from "../../../../website/src/components/Solver.js"
 
-# Day 16
+# Day 16: Packet Decoder
+
+by [@tgodzik](https://github.com/tgodzik)
 
 ## Puzzle description
 
 https://adventofcode.com/2021/day/16
 
+## Part1: You've got mail!
+
+It seems that that we can split our problem into two parts. First, we need to
+parse the example into structures that we can later use to calculate out
+results.
+
+Let's start with defining data structures to use:
+
+```scala
+enum Packet(version: Int, typeId: Int):
+  case Literal(version: Int, exprs: Long) extends Packet(version, 4)
```

Comment:
Why is it named `exprs` (plural)?

```suggestion
  case Literal(version: Int, value: Long) extends Packet(version, 4)
```

Reply:
I was reworking the examples and it got lost, thanks!